pkg/reconcile/pipeline/infinispan/handler/manage/statefulset_updates.go

```diff
@@ -84,6 +84,18 @@ func StatefulSetRollingUpgrade(i *ispnv1.Infinispan, ctx pipeline.Context) {
 		}
 	}
 
+	// Changes to probes
+	probedChanged := func(current, new *corev1.Probe) bool {
+		if reflect.DeepEqual(current, new) {
+			return false
+		}
+		*current = *new
+		return true
+	}
+	updateNeeded = probedChanged(container.LivenessProbe, provision.PodLivenessProbe(i))
+	updateNeeded = probedChanged(container.ReadinessProbe, provision.PodLivenessProbe(i))
+	updateNeeded = probedChanged(container.StartupProbe, provision.PodLivenessProbe(i))
```

Comment:
2 concernts
* all methods are using `provision.PodLivenessProbe(i)`?
* shouldn't be "or", I mean, if the startup probe is not changed, `updateNeeded` will be set to `false` independently of the other probes